Content.Shared/Storage/EntitySystems/SharedEntityStorageSystem.cs

```diff
@@ -331,7 +331,20 @@ public bool CanInsert(EntityUid toInsert, EntityUid container, SharedEntityStora
         if (component.Contents.ContainedEntities.Count >= component.Capacity)
             return false;
 
-        return CanFit(toInsert, container, component);
+        // Allow other systems to prevent inserting the item: e.g. the item is actually a ghost.
+        var attemptEvent = new InsertIntoEntityStorageAttemptEvent(toInsert);
+        RaiseLocalEvent(toInsert, ref attemptEvent);
+
+        if (attemptEvent.Cancelled)
+            return false;
+
+        // Consult the whitelist. The whitelist ignores the default assumption about how entity storage works.
+        if (component.Whitelist != null)
+            return _whitelistSystem.IsValid(component.Whitelist, toInsert);
+
+        // Mobs can be inserted into entity storage if the container isn't an item. All other entities must be an item.
+        // Being a mob has a higher priority than being an tem.
```

Comment:
Just quickly saw this.
"being an tem" > "being an item"

Please no hurt.

Reply:
that's it

i'm off to become a tibetan monk

tell my family "weh" and then refuse to elaborate